wordpress-mobile/WordPress-iOS
# Jetpack Section: Scan: Adds Scan History View

Project: #15190
Related PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/331
---

Adds scan history with error handling. 

The error messages aren't solidified yet and will need copy from @osullivanchris

### Screenshots

#### General States
| Loading | All | Fixed | Ignored |
|:---:|:---:|:---:|:---:|
|![loading](https://user-images.githubusercontent.com/793774/105534499-95650f00-5cbb-11eb-839a-eb258c86bc0c.png)|![all](https://user-images.githubusercontent.com/793774/105541893-7d46bd00-5cc6-11eb-8264-4af4668edbdf.png)|![fixed](https://user-images.githubusercontent.com/793774/105541888-7b7cf980-5cc6-11eb-9d10-f0b50c772fcd.png)|![ignored](https://user-images.githubusercontent.com/793774/105541892-7cae2680-5cc6-11eb-971a-54946e39e6a4.png)|

#### Empty States
| No History | No Fixed | No Ignored | 
|:---:|:---:|:---:|
|![nohistory](https://user-images.githubusercontent.com/793774/105534757-fab90000-5cbb-11eb-98a2-f92618b1e99d.png)|![nofixed](https://user-images.githubusercontent.com/793774/105534755-fa206980-5cbb-11eb-8527-d9d37b90aeee.png)|![noignored](https://user-images.githubusercontent.com/793774/105534753-f8ef3c80-5cbb-11eb-9363-79b704a98fa5.png)|

#### Error States
| Generic Error | No Internet |
|:---:|:---:|
|![generic](https://user-images.githubusercontent.com/793774/105535094-669b6880-5cbc-11eb-9ee0-409d2b9c0bfa.png)|![nointernet](https://user-images.githubusercontent.com/793774/105535139-761ab180-5cbc-11eb-947e-332af796b0c5.png)|

### To test:
1. Launch the app, tap on the My Site, tap on a site with Jetpack Scan
2. Tap the history item in the top right corner
3. 👁️ You should see the scan history view with a loading indicator
4. 👁️ Once it finishes loading you'll see a list of threats
5. Tap the 'Fixed' tab item
6. 👁️ You'll only see a list of fixed threats (green color)
7. Tap the 'Ignored' tab item
8. 👁️ You'll only see a list of ignored threats (gray color)
9. Pull to refresh and wait for it to complete
8. 👁️ List is auto filtered using the tab item you had selected before


### PR submission checklist:
